### ISSUE TITLE
Fix build incrementality and project dependency issues

### ISSUE THREAD

**Issue by asklar** [microsoft/react-native-windows#5148]
We have some issue with projects getting rebuilt unnecessarily as well as projects not being able to build standalone (i.e. outside a solution). We should fix this to improve dev productivity and avoid wasted dev time.

**Comment by JunielKatarn** [comment]
@NickGerleman is the incremental build issue the one related to MSBuild target inputs and outputs you mentioned?

**Comment by asklar** [comment]
@JunielKatarn no, I think that was about autolinking? This is about having two solutions open, building one, then when you build the other most of the projects end up needing to be rebuilt.
